harness/tests/determined/common/experimental/test_trial.py

```diff
@@ -93,3 +94,35 @@ def test_trial_logs_accepts_valid_timestamps(
     )
 
     list(trialref.logs(timestamp_before=None, timestamp_after=valid_timestamp))
+
+
+@mock.patch("determined.common.api.bindings.get_GetTrialCheckpoints")
+@pytest.mark.parametrize(
+    "sort_by", [checkpoint.CheckpointSortBy.SEARCHER_METRIC, "val_metric_name"]
+)
+def test_list_checkpoints_calls_bindings_with_params(
+    mock_bindings: mock.MagicMock,
+    make_trialref: Callable[[int], trial.Trial],
+    sort_by: Union[checkpoint.CheckpointSortBy, str],
+) -> None:
+    trialref = make_trialref(1)
+    ckpt_resp = api_responses.sample_get_experiment_checkpoints()
```

Comment:
The fact that `Trial.list_checkpoints` uses `api_responses.sample_get_experiment_checkpoints` tells me that one of:
* The api_responses function is improperly named
* Typing is messed up somewhere in/downstream of `Trial.list_checkpoints`
* There's a bug in this test

Reply:
yeah you're right, the api responses returns an experiment checkpoints response which is incorrect. made another method to return the proper trial checkpoints response.